applications/FluidDynamicsApplication/custom_utilities/fluid_calculation_utilities.cpp

```diff
@@ -0,0 +1,75 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// External includes
+
+// Project includes
+
+// Include base h
+#include "custom_utilities/fluid_calculation_utilities.h"
+
+namespace Kratos
+{
+
+template<>
+void FluidCalculationUtilities::AssignValue(
+    array_1d<double, 2>& rOutput,
+    const array_1d<double, 3>& rInput)
```

Comment:
Usually the inputs are put before the outputs in the signature of the functions.

Reply:
Aside of this, I'm not fully convinced about the `array_1d<double,2>...

Reply:
I can understand its not the most elegant solution (I did not have that template overload in the RANS app), but in vms_adjoint_element we are retrieving 2d velocity to `array_1d<double, 2>` and then there is a lot of `prod(dNdX, velocity)` done afterwards, which will fail if this method is used (which will need `array_1d<double, 3>` for velocity). I wanted to have a demonstrator case, and this is the only simple one I found in FluidApp. (I was thinking about changing qs_vms, but it will take more time, maybe we can change them later) :/

But the upside of this is, when we do it like that, we can use `prod(dNdx, velocity)` to get the convective operators easily in 2D and 3D without any hazzle.

I need a simple demonstrator case for this, do you have any suggestions?

Reply:
I'll change the signature :)

Reply:
signature change done :)

Reply:
About the array_1d, I agree it’s not elegant but give us more flexibility so let’s keep it for the sake of performance.

I’d appreciate the signatures changes 😉

Reply:
It is done 👍